Cargo.toml

```diff
@@ -36,4 +43,16 @@ members = [
 # Don't optimize build dependencies, like proc macros.
 # Helps with build times.
 [profile.release.build-override]
-opt-level = 0
\ No newline at end of file
+opt-level = 0
+
+# Patch `cortex-m-rt` for reasons described here:
+# https://github.com/mciantyre/teensy4-rs#runtime
+[patch.crates-io.cortex-m-rt]
+path = "cortex-m-rt-patch"
+
+# Patch `imxrt-hal` so that we may access the `hal::Peripherals::steal`
+# constructor. This branch also ensures that we use the `crates.io` instance of
+# `imxrt-ral` as the git dependency requires extra code generation steps.
+[patch.crates-io.imxrt-hal]
+git = "https://github.com/mitchmindtree/imxrt-rs"
+branch = "proto/rtic-backup"
```

Comment:
This patch is a little awkward - we might want to wait for the `steal` constructor to land in the `hal` before landing this?

Reply:
I can take care of that in the HAL crate. I'll probably bundle the `steal()` function, along with the feature-flag rename, in a 0.4.0 release.

Reply:
As of imxrt-rs/imxrt-rs#68, we should be able to use the HAL as a git dependency. I've opened a PR on that repo to add a `steal()` function. Once that lands, we can point this patch -- or the entire dependency -- to the `master` branch of the main repo.

Reply:
Thanks for the heads-up!

I've just rebased this branch onto the latest master and updated the patch to point to imxrt-rs/master and everything appears to be working nicely.